### PR TITLE
Remove camlp5 transitional

### PR DESCRIPTION
**Pull request by Zimmi48**
###### Motivation for this change

Remove annoying `camlp5_transitional` / `camlp5_strict` duality in favor of a (strict) `camlp5`. We take care of updating / removing the last bits that still require the transitional version.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

~~Currently running `now-review` but testing along the way seems to indicate that everything is fine.~~
EDIT: done.

### CODE REVIEW

**Comment by Zimmi48** [comment]
cc @Mic92 @vbgl

File: pkgs/top-level/ocaml-packages.nix [hunk 1]
```diff
@@ -674,6 +665,8 @@ let
 
     process = callPackage ../development/ocaml-modules/process { };
 
+    prooftree = callPackage  ../applications/science/logic/prooftree { };
```
[the remark this answers is not in the document]
Comment (Mic92) [reply pkgs/top-level/ocaml-packages.nix]: When we do not have it there we only have to worry about one ocaml version. You can also `ocamlPackages.callPackage` for convenience

[the remark this answers is not in the document]
Comment (Zimmi48) [reply pkgs/top-level/ocaml-packages.nix]: I see.

**Comment by Zimmi48** [comment]
Updated. I didn't rebuild everything but there should be no functional change.

**Comment by vbgl** [comment]
> You can also `ocamlPackages.callPackage` for convenience 

How do you (sanely) override a package defined like this?

**Comment by Mic92** [comment]
`override` function for packages build `ocamlPackages.callPackage` works the same as for other packages build with `callPackages` from all-packages. There is only a different scope used. So arguments are also coming from ocamlPackages.

**Comment by vbgl** [comment]
What you usually want to override, its the full `ocamlPackages` package set, which is not an argument to `callPackage`; hence my question.

**Pull request merged**